src/components/nav-logo/nav-logo.tsx

```diff
@@ -0,0 +1,96 @@
+import { Component, Element, EventEmitter, h, Host, Prop, Event } from "@stencil/core";
+import { CSS } from "./resources";
+
+@Component({
+  tag: "calcite-nav-logo",
+  styleUrl: "nav-logo.scss",
+  shadow: true
+})
+export class CalciteNavLogo {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteNavLogoElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+  /** When `true`, visually highlight the component */
+  @Prop({ reflect: true }) active: boolean;
+
+  /** Specifies the href destination of the component */
+  @Prop({ reflect: true }) href: string;
+
+  /** Specifies accesible label for the component */
+  @Prop() label: string;
+
+  /** Specifies the subtext to display, for example an organization or application description */
+  @Prop({ reflect: true }) subText: string;
+
+  /** Specifies the text to display, for example a product name */
+  @Prop({ reflect: true }) text: string;
+
+  /** When `true`, makes `text` and `subText` visible */
+  @Prop({ reflect: true }) textEnabled: boolean;
+
+  /** Specifies the `src` to an image  */
+  @Prop({ reflect: true }) thumbnail: string;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /** Emits when user select the component. */
+  @Event() calciteNavLogoSelect: EventEmitter<void>;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private clickHandler = (): void => {
+    this.calciteNavLogoSelect.emit();
+  };
+
+  private keyDownHandler = (event: KeyboardEvent): void => {
+    if (event.key === "Enter" || event.key === " ") {
+      this.calciteNavLogoSelect.emit();
+    }
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+  render() {
+    return (
+      <Host>
+        <a
+          aria-label={this.label}
+          href={this.href}
+          onClick={this.clickHandler}
+          onKeyDown={this.keyDownHandler}
+          tabIndex={0}
+        >
+          {this.thumbnail && <img src={this.thumbnail} />}
```

Comment:
should the `img` tag have a key if its conditional?

Reply:
i don't think its required in this case, using key would recreate the node which isn't required in this case.